### PR TITLE
Add Dimensions API to Data, and deprecate Filter API

### PR DESCRIPTION
**Pull request by jaredsmith**
The Data.Filter API has been deprecated, so have the Node SDK throw a deprecation warning when it gets used.

Also, add the Data.Dimensions API to the SDK.

Signed-off-by: Jared Smith <jsmith@mux.com>

### CODE REVIEW

**Review by decepulis — APPROVED** [review]
One question: Does `test/integration/nockFixtures/Dimensions/.get.json.swp` belong in this PR? 
Otherwise, best as I can tell, looks sensible!

**Comment by jaredsmith** [comment]
> Does `test/integration/nockFixtures/Dimensions/.get.json.swp` belong in this PR?

Nope, good catch.  Just removed it.

File: src/data/resources/filters.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ export class Filters extends Base {
     filterId: string,
```
Comment (cjpillsbury) [inline src/data/resources/filters.ts:42]: nitpick(non-blocker): Consider adding [`@deprecated`](https://jsdoc.app/tags-deprecated.html) to the `Filter` class definition

**Review by cjpillsbury — APPROVED** [review]
one non-blocking callout, otherwise lgtm! 😎

**Pull request merged**